### PR TITLE
Add ScaNN index param checker

### PR DESCRIPTION
**Pull request by chasingegg**
Related issue https://github.com/milvus-io/milvus/issues/27252
dim should be a multiple of 2 in scann index

### CODE REVIEW

File: pkg/util/indexparamcheck/scann_checker.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// CheckTrain checks if SCANN index can be built with the specific index parameters.
+func (c *scaNNChecker) CheckTrain(params map[string]string) error {
+	if err := c.ivfBaseChecker.CheckTrain(params); err != nil {
+		return err
+	}
+
+	return c.checkScaNNParams(params)
+}
+
+func (c *scaNNChecker) checkScaNNParams(params map[string]string) error {
```
Comment (longjiquan) [inline pkg/util/indexparamcheck/scann_checker.go:22]: Please add ut for all the corner cases.

File: pkg/util/indexparamcheck/scann_checker.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	"strconv"
+)
+
+// scaNNChecker checks if a SCANN index can be built.
+type scaNNChecker struct {
+	ivfBaseChecker
+}
+
+// CheckTrain checks if SCANN index can be built with the specific index parameters.
+func (c *scaNNChecker) CheckTrain(params map[string]string) error {
```
Comment (longjiquan) [inline pkg/util/indexparamcheck/scann_checker.go:14]: No need to use pointer receiver here.

**Review by longjiquan — CHANGES_REQUESTED** [review]

File: pkg/util/indexparamcheck/scann_checker.go (hunk 1 above)
Comment (chasingegg) [reply pkg/util/indexparamcheck/scann_checker.go:22]: scann's ut is more likely the subset of ivfpq, I guess it is enough

File: pkg/util/indexparamcheck/scann_checker.go (hunk 2 above)
Comment (chasingegg) [reply pkg/util/indexparamcheck/scann_checker.go:14]: should we change ivf_pq_checker, ivf_sq_checker as well? The code is copied here...

**Pull request merged**